src/dstack/_internal/cli/commands/run.py

```diff
@@ -89,6 +114,31 @@ def _command(self, args: argparse.Namespace):
             self._parser.print_help()
             return
 
+        termination_policy_idle = 5 * 60  # 5 minutes by default
+        termination_policy = TerminationPolicy.DESTROY_AFTER_IDLE
+
+        if args.idle_duration is not None:
+            try:
+                termination_policy_idle = int(args.idle_duration)
+            except ValueError:
+                termination_policy_idle = 60 * parse_pretty_duration(args.idle_duration)
+
+        creation_policy = (
+            CreationPolicy.REUSE if args.creation_policy_reuse else CreationPolicy.REUSE_OR_CREATE
+        )
+
+        if creation_policy == CreationPolicy.REUSE and termination_policy_idle is not None:
+            console.print(
+                "[warning]If the flag --reuse is set, the argument --idle-duration will be skipped[/]"
+            )
+            termination_policy = TerminationPolicy.DONT_DESTROY
+
+        if args.instance_name is not None and termination_policy_idle is not None:
+            console.print(
+                f"[warning]--idle-duration won't be applied to the instance {args.instance_name!r}[/]"
+            )
+            termination_policy = TerminationPolicy.DONT_DESTROY
+
```

Comment:
The pool-specific argument parsing should be refactored. See how other flags are parsed.